### ISSUE TITLE
Missing column in seo_url table

### ISSUE THREAD

**Issue by githubertus** [opencart/opencart#8001]
**What version of OpenCart are you reporting this for?**
3.0.3.4-core-pre
**Describe the bug**
trying to edit a product gives

`Fatal error: Uncaught Exception: Error: Unknown column 'query' in 'where clause'<br />Error No: 1054<br />SELECT * FROM oc_seo_url WHERE query = 'product_id=42' in system/library/db/mysqli.php:40

 Stack trace: #0 system/library/db.php(45): DB\MySQLi->query('SELECT * FROM o...') #1 admin/model/catalog/product.php(595): DB->query('SELECT * FROM o...') #2 system/engine/loader.php(248): ModelCatalogProduct->getSeoUrls('42') #3 system/engine/proxy.php(47): Loader->{closure}(Array, Array) #4 admin/controller/catalog/product.php(1153): Proxy->__call('getSeoUrls', Array) #5 admin/controller/catalog/product.php(116): ControllerCatalogProduct->getForm() #6 system/engine/action.php(79): ControllerCatalogProduct->edit() #7 admin/controller/startup/router.php(26): Action->execute(Object(R in system/library/db/mysqli.php on line 40
`

**Comment by straightlight** [comment]
This issue has now been fixed in the pre-release version. Therefore, this commit will now be closed.

**Comment by githubertus** [comment]
query column still not available in 3.0.3.4-core-pre release update

**Comment by straightlight** [comment]
Already explained on the forum: https://forum.opencart.com/viewtopic.php?f=201&t=218203&start=240#p790912 .

**Comment by cgcmart** [comment]
Merged or updated seo changes don't work. See closed issue without solution [https://github.com/opencart/opencart/issues/7942](url)

**Comment by straightlight** [comment]
@cgcmart: Fixed link on the above. However, the provided solution on that commit location is still unconfimed.

**Comment by cgcmart** [comment]
seo changes need more work. MySQL says column name "key" is reserved for "keyword". Why need it replace "query"?

**Comment by straightlight** [comment]
> seo changes need more work. MySQL says column name "key" is reserved for "keyword". Why need it replace "query"?

Please provide the location where it states a reserved word for query. More information is needed.

**Comment by cgcmart** [comment]
An old open issue [#7660](https://github.com/opencart/opencart/issues/7660)

> Notice: Undefined index: key in

[#7662](https://github.com/opencart/opencart/issues/7662)

> Table oc_seo_regex have no key field

when the issue raise, I tried to add "key" field manually by phpMyAdmin, that gave warning name "key" is reserved for "keyword" by MySQL. I use MariaDB 10.3. It lets you add the field, but seo still not working.

After that trial failure, I commented out "key" field from all related seo files, and modified seo files, my master branch site seo completely works. I did not update any recent seo changes from oc merges.
